### PR TITLE
feat(opentrons-ai-client): Created an update protocol page

### PR DESCRIPTION
**Pull request by connected-znaim**
…ol page basic layout

# Overview

Added a new page that allows us to upload an existing protocol for modification. It has error states for file size is zero and when uploading a non-python file.

Drop down options are:
const updateOptions: DropdownOption[] = [
  {
    name: 'Adapt Python protocol from OT-2 to Flex',
    value: 'adapt_python_protocol',
  },
  { name: 'Change labware', value: 'change_labware' },
  { name: 'Change pipettes', value: 'change_pipettes' },
  { name: 'Other', value: 'other' },
]

START STATE:
image

EMPTY FILE:
image

NON-PYTHON FILE:
image

ALL FIELDS FILLED OUT:
image

## Test Plan and Hands on Testing

Tested non python files and empty python files. Made sure the prompt text is correctly modified to include the fields filled out.

### CODE REVIEW

File: opentrons-ai-client/src/molecules/UploadInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          {typeof uploadText === 'string' ? (
+            <LegacyStyledText
+              as="p"
+              textAlign={TYPOGRAPHY.textAlignCenter}
+              marginTop={SPACING.spacing16}
+            >
+              {uploadText}
+            </LegacyStyledText>
+          ) : (
+            <>{uploadText}</>
```
Comment (fbelginetw) [inline opentrons-ai-client/src/molecules/UploadInput/index.tsx:125]: I think we don't need a Fragment here

File: opentrons-ai-client/src/organisms/UpdateProtocol/index.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+      setFile(file)
+    } else {
+      setErrorText(t('python_file_type_error'))
+      setFile(file)
+    }
+  }
+
+  function processDataAndNavigateToChat(): void {
+    const userPrompt = `Modify the following Python code using the Opentrons Python Protocol API v2. Ensure that the new labware and pipettes are compatible with the Flex robot. Ensure that you perform the correct Type of Update use the Details of Changes.
```
Comment (fbelginetw) [inline opentrons-ai-client/src/organisms/UpdateProtocol/index.tsx]: I think this should be in the translation file

File: opentrons-ai-client/src/molecules/UploadInput/index.tsx (hunk 1 above)
Comment (connected-znaim) [reply opentrons-ai-client/src/molecules/UploadInput/index.tsx:125]: This is their existing component copied over temporarily.

File: opentrons-ai-client/src/organisms/UpdateProtocol/index.tsx (hunk 2 above)
Comment (connected-znaim) [reply opentrons-ai-client/src/organisms/UpdateProtocol/index.tsx]: done

**Review by fbelginetw — APPROVED** [review]
There's some conflicts, but approved!

**Pull request merged**